pandas/tests/window/test_rolling.py

```diff
@@ -1871,3 +1871,18 @@ def test_rolling_skew_kurt_floating_artifacts():
     assert (result[-2:] == 0).all()
     result = r.kurt()
     assert (result[-2:] == -3).all()
+
+
+def test_rolling_imaginary_part_of_complex(arithmetic_win_operators):
+    # GH 46619
+    func_name = arithmetic_win_operators
+    df = DataFrame([1j, 1 + 2j])
+    result = getattr(
+        df.rolling(2).apply(lambda x: print(x) is None),
```

Comment:
So `func_name` should now be tested instead of `apply`

Reply:
Hi Sorry. I not able to understand fully. If I not use apply.
```
func_name = arithmetic_win_operators
df = DataFrame([1j, 1 + 2j])
result = getattr(
    df.rolling(2),
    func_name,
)()
```

Will  get AssertionError depend function from `arithmetic_win_operators`

Reply:
@mroeschke 

The code I run 
'''
func_name = arithmetic_win_operators
df = DataFrame([1j, 1 + 2j])
result = getattr(
    df.rolling(2),
    func_name,
)()
'''
The 'arithmetic_win_operators' can be sum, mean, median, max, min, var, std, kurt, skew, count, sem (11 possible arithmetic function. 

For example, if arithmetic_win_operators='sum', the error is  raised that function sum does not support complex
dtypes in cython function in 'pandas/_libs/window/aggregations.pyx'
```
Traceback (most recent call last):
  File "/home/open_source/pandas_2/development/development_5.py", line 26, in <module>
    result = getattr(
  File "/home/open_source/pandas_2/pandas/pandas/core/window/rolling.py", line 1974, in sum
    return super().sum(*args, engine=engine, engine_kwargs=engine_kwargs, **kwargs)
  File "/home/open_source/pandas_2/pandas/pandas/core/window/rolling.py", line 1407, in sum
    return self._apply(window_func, name="sum", **kwargs)
  File "/home/open_source/pandas_2/pandas/pandas/core/window/rolling.py", line 619, in _apply
    return self._apply_blockwise(homogeneous_func, name)
  File "/home/open_source/pandas_2/pandas/pandas/core/window/rolling.py", line 487, in _apply_blockwise
    res = hfunc(arr)
  File "/home/open_source/pandas_2/pandas/pandas/core/window/rolling.py", line 477, in hfunc
    return homogeneous_func(values)
  File "/home/open_source/pandas_2/pandas/pandas/core/window/rolling.py", line 614, in homogeneous_func
    result = calc(values)
  File "/home/open_source/pandas_2/pandas/pandas/core/window/rolling.py", line 611, in calc
    return func(x, start, end, min_periods, *numba_args)
  File "pandas/_libs/window/aggregations.pyx", line 132, in pandas._libs.window.aggregations.roll_sum
ValueError: Buffer dtype mismatch, expected 'const float64_t' but got 'complex float'
```

Reply:
- [ ] sum
- [ ] mean
- [ ] median
- [ ] max
- [ ] min
- [ ] var
- [ ] std
- [ ] kurt
- [ ] skew
- [ ] count
- [ ] sem

Reply:
@mroeschke Hi
I not sure whether I do it correctly. I try to update `roll_mean` to support complex type in this file `pandas/_libs/window/aggregations.pyx` 

'''
ctypedef fused float_complex_t:
    float64_t
    complex64_t

cdef inline void add_mean(float_complex_tval, Py_ssize_t *nobs, float_complex_t*sum_x,
                          Py_ssize_t *neg_ct, float_complex_t*compensation,
                          int64_t *num_consecutive_same_value, float_complex_t*prev_value) nogil:
    """ add a value from the mean calc using Kahan summation """
    cdef:
        float_complex_ty, t

    # Not NaN
    if val == val:
        nobs[0] = nobs[0] + 1
        y = val - compensation[0]
        t = sum_x[0] + y
        compensation[0] = t - sum_x[0] - y
        sum_x[0] = t
        if signbit(val):
            neg_ct[0] = neg_ct[0] + 1

        # GH#42064, record num of same values to remove floating point artifacts
        if val == prev_value[0]:
            num_consecutive_same_value[0] += 1
        else:
            # reset to 1 (include current value itself)
            num_consecutive_same_value[0] = 1
        prev_value[0] = val
```

```
Error compiling Cython file:
------------------------------------------------------------
...
        nobs[0] = nobs[0] + 1
        y = val - compensation[0]
        t = sum_x[0] + y
        compensation[0] = t - sum_x[0] - y
        sum_x[0] = t
        if signbit(val):
                  ^
------------------------------------------------------------

pandas/_libs/window/aggregations.pyx:231:19: Cannot assign type 'float complex' to 'long double'
```

Reply:
Will probably need cast this to float before passing in this C func, or just use a comparison operator to replace sigbit.

Reply:
I do not modify code in cython pandas/_libs/window/aggregations.pyx file to support complex type. Do cast the values from complex to float before passing it to C func.